karadoc/common/table_utils.py

```diff
@@ -53,12 +53,9 @@ def parse_table_name(table_name: str) -> Tuple[str, str, str]:
         raise ValueError("Incorrect table name : %s" % table_name)
 
 
-def table_exists(table_name: str, job_type: Type[JobBase]) -> bool:
-    folder_location = get_folder_location(table_name, job_type)
-    if folder_location:
-        return os.path.isdir(folder_location)
-    else:
-        return False
+def table_exists(table_name: str) -> bool:
+    (current_schema_name, current_table_name, _) = parse_table_name(table_name)
+    return any([(current_schema_name == s and current_table_name == t) for (s, t, _) in list_schema_table_folders()])
```

Comment:
No. You replaced a call to `get_folder_location` which is in O(1) with a call to `list_schema_table_folders` which is in O(n), this will result in O(n^2) performances on some operations, which will slow down validation commands on large projects.

Reply:
Suggestion:

Use get_folder_location without any job_type and apply the following changes:


```
def get_folder_location(table_name: str) -> Optional[str]:
    (schema_name, table_name, _) = parse_table_name(table_name)

    path = file_index.get_action_file(schema_name, table_name)
    if path:
        return os.path.split(path)[0]
    else:
        return None

def get_action_file(schema_name: str, table_name: str, job_type: Type[JobBase] = None) -> Optional[str]:
    """Recursively scan the model folder and return the path of the ACTION_FILE.py file for the given table_name
    :param schema_name: name of the schema
    :param table_name: name of the table
    :param job_type: job type defined in the action file. (e.g. SparkBathJob, QualityCheckJob ...)
    :return: A string path
    """
    if job_type is None:
        action_file_name = None
    else:
        assert_true(issubclass(job_type, JobBase))
        action_file_name = job_type.get_action_file_name()
    result = __get_indexed_action_file(action_file_name).get(schema_name, {}).get(table_name)
    if result:
        return result
    else:
        return None


# We use a cache here to avoid recomputing everything all the time.
@lru_cache(maxsize=100)
def __index_files_with_name_cached(file_name: Optional[str]) -> Dict[str, Dict[str, str]]:
    """Recursively scan the model folder and returns all the relative paths of the files
    indexed by schema and table names.
    If file_name is None, return the path of the model folders that contain at least one file.

    :param file_name: name of the files to match
    :return:
    """
    schemas = {}
    for schema, table, folder in list_schema_table_folders():
        for file in os.listdir(folder):
            if file_name is None:
                schemas.setdefault(schema, {})[table] = folder
                break
            if file == file_name:
                schemas.setdefault(schema, {})[table] = os.path.join(folder, file)

    return schemas

```

Reply:
you're right, following this implementation decreased the execution time of our pipelines from 8 minutes to 2 minutes.
I updated this MR